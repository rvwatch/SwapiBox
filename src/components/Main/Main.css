.main-wrap{
    position: relative;
    z-index: 4;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
}

.main-wrap section {
    width: 75%;
}