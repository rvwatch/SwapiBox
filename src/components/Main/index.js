import React, { Component } from 'react';
import Home from '../Home';
import People from '../People';
import Planets from '../Planets';
import Vehicles from '../Vehicles';
import Favorites from '../Favorites';
import { Route, Switch } from 'react-router-dom';


class Main extends Component {
  constructor(){
    super();
    this.state = {
      favorites: []
    };
  }

  setFavorites = (favorite) => {
    console.log(favorite)
    this.setState({
      favorites: {}
    });
    console.log(this.state.favorites);
    
  }

  render() {
    return (
      <div className={'main-wrap'}>

        <Switch>
          <Route exact path="/" component={Home}/>
          <Route path='/People'  render={() => (<People setFavorites={this.setFavorites} favorites={this.state.favorites} />)} />
          <Route path='/Planets' render={() => (<Planets setFavorites={this.setFavorites} favorites={this.state.favorites} />)} />
          <Route path='/Vehicles' render={() => (<Vehicles setFavorites={this.setFavorites} favorites={this.state.favorites} />)} />
          <Route path='/Favorites' render={() => (<Favorites setFavorites={this.setFavorites} favorites={this.state.favorites} />)} />
        </Switch>
      </div>
    );
  }
}

export default Main;