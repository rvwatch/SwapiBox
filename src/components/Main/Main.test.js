import React from 'react';
import { shallow } from 'enzyme';
import Main from './index';

describe('Main ', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Main />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

});