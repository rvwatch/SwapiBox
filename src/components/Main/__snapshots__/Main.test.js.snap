// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main  renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
                component={[Function]}
                exact={true}
                path="/"
        />
        <Route
                path="/People"
                render={[Function]}
        />
        <Route
                path="/Planets"
                render={[Function]}
        />
        <Route
                path="/Vehicles"
                render={[Function]}
        />
        <Route
                path="/Favorites"
                render={[Function]}
        />
</Switch>,
      "className": "main-wrap",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            path="/People"
            render={[Function]}
/>,
          <Route
            path="/Planets"
            render={[Function]}
/>,
          <Route
            path="/Vehicles"
            render={[Function]}
/>,
          <Route
            path="/Favorites"
            render={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/People",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/Planets",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/Vehicles",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/Favorites",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    path="/People"
                    render={[Function]}
          />
          <Route
                    path="/Planets"
                    render={[Function]}
          />
          <Route
                    path="/Vehicles"
                    render={[Function]}
          />
          <Route
                    path="/Favorites"
                    render={[Function]}
          />
</Switch>,
        "className": "main-wrap",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              path="/People"
              render={[Function]}
/>,
            <Route
              path="/Planets"
              render={[Function]}
/>,
            <Route
              path="/Vehicles"
              render={[Function]}
/>,
            <Route
              path="/Favorites"
              render={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/People",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/Planets",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/Vehicles",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/Favorites",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
